src/core/machine-id-setup.c

```diff
@@ -180,70 +187,99 @@ int machine_id_setup(const char *root, sd_id128_t machine_id, sd_id128_t *ret) {
 
 int machine_id_commit(const char *root) {
         _cleanup_close_ int fd = -1, initial_mntns_fd = -1;
-        const char *etc_machine_id;
+        const char *etc_machine_id, *run_machine_id;
         sd_id128_t id;
         int r;
 
-        /* Replaces a tmpfs bind mount of /etc/machine-id by a proper file, atomically. For this, the umount is removed
-         * in a mount namespace, a new file is created at the right place. Afterwards the mount is also removed in the
-         * original mount namespace, thus revealing the file that was just created. */
-
         etc_machine_id = prefix_roota(root, "/etc/machine-id");
+        run_machine_id = prefix_roota(root, "/run/machine-id");
 
         r = path_is_mount_point(etc_machine_id, NULL, 0);
-        if (r < 0)
+        if (r < 0 && errno != ENOENT)
                 return log_error_errno(r, "Failed to determine whether %s is a mount point: %m", etc_machine_id);
-        if (r == 0) {
-                log_debug("%s is not a mount point. Nothing to do.", etc_machine_id);
+        if (r == 0 || ( r < 0 && errno == ENOENT) ) {
```

Comment:
Instead of this ugly repeated condition, maybe reverse the order of the branches
```C
if (r > 0) {
} else {
}
```
?

Reply:
will do